pkg/tasks/python.go

```diff
@@ -27,13 +28,54 @@ func parserPython(config *taskConfig, task *Task) error {
 	task.featureName = "python"
 	task.featureParam = version
 
+	task.addAction(&pyenvPath{})
+	task.addAction(&pyenv{})
 	task.addAction(&pythonPyenv{version: version})
 	task.addAction(&pythonInstallVenv{version: version})
 	task.addAction(&pythonCreateVenv{version: version})
 
 	return nil
 }
 
+type pyenvPath struct{}
+
+func (p *pyenvPath) description() string {
+	return ""
+}
+
+func (p *pyenvPath) needed(ctx *context) (bool, error) {
+	match, _ := regexp.MatchString("\\.pyenv/bin", ctx.env.Get("PATH"))
+	return !match, nil
+}
+
+func (p *pyenvPath) run(ctx *context) error {
+	ctx.ui.TaskWarning("The PATH environment variable should have 'pyenv' in it. Try adding the following to ~/.bashrc\n\nexport PATH=\"~/.pyenv/bin:$PATH\"")
+	return nil
+}
+
+type pyenv struct{}
+
+func (p *pyenv) description() string {
+	return "install PyEnv"
+}
+
+func (p *pyenv) needed(ctx *context) (bool, error) {
+	_, err := helpers.NewPyEnv()
```

Comment:
For some reason even after installing pyenv and adding it to my `PATH` this is still returning the error `exec: "pyenv": executable file not found in $PATH`. If I run `pyenv root` (which is what that is running underneath) I get back `/Users/jduff/.pyenv` which seems like it should be successful. Any pointers here would be appreciated.

Reply:
Probably not very helpful in this case, but related to debugging: you can get debug logging with the shell commands: `bud-enable-debug` / `bud-disable-debug`.

Here is the output of `pyenv init -`:

```
export PATH="/Users/pior/.pyenv/shims:${PATH}"
export PYENV_SHELL=bash
source '/usr/local/Cellar/pyenv/1.2.7/libexec/../completions/pyenv.bash'
command pyenv rehash 2>/dev/null
pyenv() {
  local command
  command="${1:-}"
  if [ "$#" -gt 0 ]; then
    shift
  fi

  case "$command" in
  rehash|shell)
    eval "$(pyenv "sh-$command" "$@")";;
  *)
    command pyenv "$command" "$@";;
  esac
}
```

I think we need `.pyenv/shims` in PATH (contrary to what the README mentions at one point).

Reply:
Hm, I just tried that and it didn't work. If that is the only thing I put onto my path `pyenv` doesn't work either. Something interesting is `which pyenv` doesn't return anything either, not sure if that is relevant at all.

I tried running what `LookPath` would do based on this https://golang.org/src/os/exec/lp_unix.go?s=924:966#L24 in a go shell and it seems like it's okay with that file :/.

Reply:
With the change `/bin` -> `/shims` it seems to work on my machine which has `/Users/pior/.pyenv/shims` in $PATH

Reply:
Did you install pyenv with homebrew or using the installer script? I’m wondering if there could be some difference in how homebrew sets things up.

An alternative would be to direct the user to install pyenv themselves. For whatever reason that still doesn’t work for me when using the installer script though.

I’ll try installing with homebrew and see if it works any different for me.

Reply:
Installing with homebrew seems to be the way to go, it sets up the path with `.pyenv/shims` as well so we don't need the task to warn the user about that.